File: Python/linkedlist.py
class Node(object):
    """

    Node for linked list object.

    """
    def __init__(self, data, prev=None, next=None):
        self.data = data
        self.prev = prev
        self.next = next


class LinkedList(object):
    """

    Implementation of a Doubly linked list.

    """
    def __init__(self, data):
        self.head = Node(data)

    def search(self, data):
        """

        Search for data in linked list.

        """
        curr = self.head
        while curr is not None and curr.data != data:
            curr = curr.next
        return curr

    def last_node(self):
        """

        Return the last node in the linked list.

        """
        curr = self.head
        while curr.next is not None:
            curr = curr.next
        return curr

    def insert(self, data):
        """

        Insert data into linked list.

        """
        new_node = Node(data)
        last_node = self.last_node()
        last_node.next = new_node
        new_node.prev = last_node
        return

    def delete(self, data):
        """

        Delete data from the linked list.

        """
        curr = self.search(data)
        if curr is None:
            # data not found in list.
            return
        curr.prev.next = curr.next
        curr.next.prev = curr.prev
        curr = None
        return data

    def print_linkedlist(self):
        curr = self.head
        while curr is not None:
            print curr.data
            curr = curr.next
        return

if __name__ == "__main__":
    test = LinkedList(1)
    test.insert(2)
    test.insert(3)
    test.insert(4)

    test.delete(3)
    test.print_linkedlist()

File: PHP/data_structures/linked_lists/linked_list.php
<?php

/**
 * Implementation of a basic linked list.
 */
class Node {
    private $data;
    private $next;

    public function __construct($data, &$next=NULL) {
        $this->data = $data;
        $this->next = $next;
    }

    public function getData() {
        return $this->data;
    }

    public function getNext() {
        return $this->next;
    }

    public function setNext($node) {
        $this->next = $node;
    }
}

class LinkedList {
    private $head;

    public function __construct($data) {
        $this->head = new Node($data);
    }

    /**
     * Insert a new node into list.
     * Takes O(n) time.
     */
    public function insert($data) {
        $newNode = new Node($data);
        $curr = $this->head;
        while ($curr->getNext() != NULL) {
            $curr = $curr->getNext();
        }
        $curr->setNext($newNode);
    }

    public function search($data) {
        $curr = $this->head;
        $result = "{$data} not found." . PHP_EOL;
        $pos = 0;
        while ($curr != NULL) {
            if ($curr->getData() == $data) {
                $result = "{$data} found at position " . $pos . PHP_EOL; 
                break;
            }
            $curr = $curr->getNext();
            $pos++;
        }
        return $result;
    }

    public function delete($data) {
        // handle case where data is head node
        if ($this->head->getData() == $data) {
            $delNode = $this->head;
            $this->head = $this->head->getNext();
            unset($delNode);
            return "{$data} deleted." . PHP_EOL;
        } else if(!$this->hasData($data)) {
            return "Data {$data} not found in list." . PHP_EOL;
        }        
        // handle case where data is mid/last node
        $prevNode = $this->getPrevNode($data);
        $delNode = $prevNode->getNext();
        $prevNode->setNext($delNode->getNext());
        unset($delNode);
        return "{$data} deleted." . PHP_EOL;
    }

    public function getHead() {
        return $this->head;
    }

    public function printList() {
        $curr = $this->head;
        while ($curr != NULL) {
            echo $curr->getData() . "->";
            $curr = $curr->getNext();
        }
        echo "NULL" . PHP_EOL;
    }

    private function getPrevNode($data) {
        $prev = $this->head;
        $curr = $this->head->getNext();
        while ($curr != NULL) {
            if ($curr->getData() == $data) {
                return $prev;
            }
            $prev = $curr;
            $curr = $curr->getNext();
        }
        return NULL;
    }

    private function hasData($data) {
        $curr = $this->head;
        while ($curr != NULL) {
            if ($curr->getData() == $data) {
                return true;
            }
            $curr = $curr->getNext();
        }
        return false;
    }
}


$linkedList = new LinkedList(1);
$linkedList->insert(2);
$linkedList->insert(3);
$linkedList->insert(4);
$linkedList->insert(5);
$linkedList->insert(6);
$linkedList->printList();
echo $linkedList->search(1);
echo $linkedList->search(3);
echo $linkedList->search(6);
echo $linkedList->search(8);
echo $linkedList->delete(4);
$linkedList->printList();